roles/oneagent/tests/component/run.py

```diff
@@ -0,0 +1,167 @@
+import argparse
+import glob
+import logging
+import os
+import shutil
+import subprocess
+import sys
+from datetime import datetime
+
+from pathlib import Path
+from typing import Any
+from scripts.util.test_data_types import DeploymentPlatform
+
+USER_KEY = "user"
+PASS_KEY = "password"
+
+BASE_DIR = Path(__file__).resolve().parent
+TEST_DIR = BASE_DIR / "test_dir"
+LOG_DIR = TEST_DIR / "logs"
+INSTALLERS_DIR = TEST_DIR / "installers"
+TEST_VARS = {"PYTHONPATH": "scripts/"}
+
+
+class ServerWrapper(object):
+    def __init__(self, proc: subprocess.Popen):
+        self.proc = proc
+
+    def __enter__(self):
+        return self.proc
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.proc.terminate()
+        save_log(self.proc.stdout, LOG_DIR / "server.log")
+
+
+def get_env_vars() -> dict[str, str]:
+    # This is required to pass current venv vars down to the subprocess for tests and server
+    env_vars = os.environ.copy()
+    env_vars.update(TEST_VARS)
+    return env_vars
+
+
+def save_log(out, log_path: Path) -> None:
+    with log_path.open("w") as log:
+        for line in out:
+            log.write(line)
+
+
+def get_test_args(args: dict[str, Any]) -> list[str]:
+    test_args = [f"--{USER_KEY}='{args[USER_KEY]}'", f"--{PASS_KEY}='{args[PASS_KEY]}'"]
+    for platform in [p.value for p in DeploymentPlatform]:
+        hosts = ','.join(args[platform])
+        hosts and test_args.append(f"--{platform}={hosts}")
+    return test_args
+
+
+def run_test(test: str, test_args: list[str]) -> bool:
+    test_name = Path(test).stem
+    logging.info(f"Test: {test_name}")
+    proc = subprocess.run(["pytest", test] + test_args, env=get_env_vars(), encoding="utf-8",
```

Comment:
please do not call pytest as subprocess. Pytest have all you need to configure and run the tests